From 33e465e5c9b438ea1a96c6320b2840e2a0f11966 Mon Sep 17 00:00:00 2001
From: Vedant Yevale <vyevale@qti.qualcomm.com>
Date: Fri, 22 Aug 2025 14:32:54 +0530
Subject: [PATCH] disp: msm: dsi: Nullify display modes after kfree

The 'display->modes' pointer must be set to NULL immediately
after calling kfree on it to prevent potential double-free
vulnerabilities or use-after-free issues.

This change ensures robust memory management by clearing the
pointer to previously freed memory, aligning with best practices
for kernel memory deallocation.

Change-Id: I4fd939bc6ee5f3d9c506f0c311a9400f366e7fd2
Signed-off-by: Vedant Yevale <vyevale@qti.qualcomm.com>
(cherry picked from commit 1d3c5a477ea8f6aca1158209e1b87209c21c61f5)
---
 techpack/display/msm/dsi/dsi_display.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/techpack/display/msm/dsi/dsi_display.c b/techpack/display/msm/dsi/dsi_display.c
index 506d24720..5419087a0 100644
--- a/techpack/display/msm/dsi/dsi_display.c
+++ b/techpack/display/msm/dsi/dsi_display.c
@@ -7826,8 +7826,10 @@ int dsi_display_get_modes(struct dsi_display *display,
 	rc = 0;
 
 error:
-	if (rc)
+	if (rc) {
 		kfree(display->modes);
+		display->modes = NULL;
+	}
 
 	mutex_unlock(&display->display_lock);
 	return rc;
-- 
2.43.0

